test/extended/router/shard/shard.go

```diff
@@ -32,20 +30,41 @@ var ingressControllerNonDefaultAvailableConditions = []operatorv1.OperatorCondit
 	{Type: "Admitted", Status: operatorv1.ConditionTrue},
 }
 
-func DeployNewRouterShard(oc *exutil.CLI, timeout time.Duration, cfg Config) (string, error) {
-	jsonCfg, err := oc.AsAdmin().Run("process").Args("-f", cfg.FixturePath, "-p",
-		"NAMESPACE=openshift-ingress-operator",
-		"DOMAIN="+cfg.Domain,
-		"TYPE="+cfg.Type).OutputToFile("config.json")
-	if err != nil {
-		return "", err
+func DeployNewRouterShard(oc *exutil.CLI, timeout time.Duration, cfg Config) (*operatorv1.IngressController, error) {
+	ingressCtrl := &operatorv1.IngressController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cfg.Type,
+			Namespace: "openshift-ingress-operator",
+			Annotations: map[string]string{
+				"ingress.operator.openshift.io/default-enable-http2": "true",
+			},
+		},
+		Spec: operatorv1.IngressControllerSpec{
+			Replicas: utilpointer.Int32(1),
+			Domain:   cfg.Domain,
+			EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+				Type: operatorv1.LoadBalancerServiceStrategyType,
+			},
+			NodePlacement: &operatorv1.NodePlacement{
+				NodeSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"node-role.kubernetes.io/worker": "",
+					},
+				},
+			},
+			NamespaceSelector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"type": cfg.Type,
+				},
+			},
+		},
 	}
-
-	if err := oc.AsAdmin().Run("create").Args("-f", jsonCfg, "--namespace=openshift-ingress-operator").Execute(); err != nil {
-		return "", err
+	_, err := oc.AdminOperatorClient().OperatorV1().IngressControllers(ingressCtrl.Namespace).Create(context.Background(), ingressCtrl, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
 	}
 
-	return jsonCfg, waitForIngressControllerCondition(oc, timeout, types.NamespacedName{Namespace: "openshift-ingress-operator", Name: oc.Namespace()}, ingressControllerNonDefaultAvailableConditions...)
+	return ingressCtrl, waitForIngressControllerCondition(oc, timeout, types.NamespacedName{Namespace: "openshift-ingress-operator", Name: oc.Namespace()}, ingressControllerNonDefaultAvailableConditions...)
```

Comment:
@Miciah shold not `Name: oc.Namespace()` by rather `Name: ingressCtrl.Name`?

Reply:
You're right.  The current code works because the test configures `cfg.Type` to be `oc.Namespace()` and the template sets the ingresscontroller's name to `cfg.Type` too, but it is probably entirely by accident that this all lines up.  Feel free to add a commit to change `Name: oc.Namespace()` to `Name: ingressCtrl.Name`.

Reply:
Done